modules/objdetect/include/opencv2/objdetect/generic_graphical_code.hpp

```diff
@@ -0,0 +1,93 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html
+#ifndef OPENCV_OBJDETECT_GENERIC_GRAPHICAL_CODE_HPP
+#define OPENCV_OBJDETECT_GENERIC_GRAPHICAL_CODE_HPP
+
+#include <opencv2/core.hpp>
+
+namespace cv {
+
+//! @addtogroup objdetect_common
+//! @{
+
+class CV_EXPORTS_W_SIMPLE GenericGraphicalCode {
+public:
+    CV_DEPRECATED_EXTERNAL  // avoid using in C++ code, will be moved to "protected" (need to fix bindings first)
+    GenericGraphicalCode();
+
+    GenericGraphicalCode(const GenericGraphicalCode&) = default;
+    GenericGraphicalCode(GenericGraphicalCode&&) = default;
+    GenericGraphicalCode& operator=(const GenericGraphicalCode&) = default;
+    GenericGraphicalCode& operator=(GenericGraphicalCode&&) = default;
+
+    /** @brief Detects graphical code in image and returns the quadrangle containing the code.
+     @param img grayscale or color (BGR) image containing (or not) graphical code.
+     @param points Output vector of vertices of the minimum-area quadrangle containing the code.
+     */
+    CV_WRAP bool detect(InputArray img, OutputArray points) const;
+
+    /** @brief Decodes graphical code in image once it's found by the detect() method.
+
+     Returns UTF8-encoded output string or empty string if the code cannot be decoded.
+     @param img grayscale or color (BGR) image containing graphical code.
+     @param points Quadrangle vertices found by detect() method (or some other algorithm).
+     @param straight_code The optional output image containing binarized code, will be empty if not found.
+     */
+    CV_WRAP std::string decode(InputArray img, InputArray points, OutputArray straight_qrcode = noArray()) const;
```

Comment:
> straight_qrcode

Name itself says that it should go to `QRCodeDetectorBase::` special methods.

Here we should have minimal methods with minimal parameters which cover the most part of use generic use cases.

Reply:
parameter "straight_qrcode" will be renamed to "straight_code"
`straight code` also exists for barcodes, aruco codes and any binary codes
The main problem is the destruction of compatibility if this parameter is removed.

Reply:
Interpretation of this value is class-specific.
In generic way it is just unusable.
Methods with that parameter should go to class-specific interfaces.

Reply:
`decode` with the `straight_code` in `QRCodeDetector` parameter and `decode` without the `straight_code` parameter in `GraphicalCodeDetector` cannot exist together.